swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -433,11 +439,36 @@ extension WorkflowNode.SubtreeManager {
             self.outputMap = outputMap
             super.init(eventPipe: eventPipe)
 
+            let signpostRef = NSObject()
+
+            if #available(iOS 12.0, *) {
+                let signpostID = OSSignpostID(log: .worker, object: signpostRef)
+                os_signpost(.begin, log: .worker, name: "Worker Created", signpostID: signpostID, "Worker: %{public}@", String(describing: W.self))
+            }
+
             signalProducer
                 .take(during: lifetime)
                 .observe(on: QueueScheduler.workflowExecution)
-                .startWithValues { [weak self] output in
-                    self?.handle(output: output)
+                .start { [weak self] event in
+                    switch event {
+                    case .value(let output):
+                        self?.handle(output: output)
```

Comment:
Does it make sense to log an event to indicate a worker emitted a value and it's about to be handled?

Reply:
Maybe move the logging to a different function. `logWorkerState(event:)`?

Reply:
I'm exploring using `.on(…)` in the stream (similar to what `.logEvents` does). I'll update this soon.

Reply:
Did the `.on(..)` work out?